src/shared/lib/fetch/index.js

```diff
@@ -162,17 +162,29 @@ export const pdf = async (scraper, url, cacheKey = 'default', date, options) =>
   return data;
 };
 
-const fetchHeadless = async url => {
+/**
+ * Load a url in headless browser, call user-supplied callback, return callback's output
+ * @param {*} scraper the scraper object
+ * @param {string} url URL of the resource
+ * @param {null} cacheKey The cache key -- this is IGNORED by this function as data is dynamically extrated and cannot be cached
+ * @param {function} callback The callback to be used on open page. Should accept a single argument `page` that is a puppeteer page that has been loaded with the given url
+ */
+export const fetchHeadlessCallback = async (scraper, url, callback, cacheKey = null) => {
```

Comment:
Though it's not obvious, we prefer to use the methods `headless`, `page`, etc, rather than widening that api to include things like this.  I believe we can accommodate your aim by allowing a "callback" option, working on that.

Reply:
This idea will also be easier to port to Li.